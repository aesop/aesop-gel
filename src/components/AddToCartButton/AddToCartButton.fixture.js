/* istanbul ignore file */
export default {
  className: '',
  copy: {
    cartAction: 'Add to your cart',
    updateNotification: 'added to your cart',
    outOfStock: {
      label: 'Out of Stock',
      title: 'Out of stock',
    },
  },
  dataTestRef: 'DATA_TEST_REF',
  isEnabled: true,
  isFullWidth: true,
  productName: 'Lorem ipsum dolor',
  theme: 'dark',
};
