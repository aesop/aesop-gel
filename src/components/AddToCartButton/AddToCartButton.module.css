@import 'styles/index';

.base {
  width: min-content;
  min-width: auto;
  max-width: 100%;
  justify-content: center;
  padding: 20px 15px 20px;
  font-size: rem(14px);
  text-align: center;

  @media (--viewport-lg) {
    max-width: 470px;
  }
}

.fullWidth {
  width: 100%;
}

.loading {
  position: absolute;
  top: calc(50% - 3.5px);
  left: 0;
}

.showSuccessMessage {
  span {
    animation: slide-up 2000ms var(--easing-ease-in-out-cubic) forwards;
  }
}

.label {
  display: block;
  overflow: hidden;
  height: 16px;
  line-height: 1;
  opacity: 1;
  transition: opacity 400ms var(--easing-ease-in-out-cubic);
  vertical-align: middle;

  span {
    display: block;
    overflow: hidden;
    height: 18px;
    margin-right: auto;
    margin-left: auto;
    line-height: 1;
    text-overflow: ellipsis;
    transform: translateY(-18px);
    transition: all 300ms var(--easing-ease-in-out-cubic);
    vertical-align: middle;
    white-space: nowrap;
  }
}

.hideLabel {
  opacity: 0;
}

@keyframes slide-up {
  0% {
    transform: translateY(0);
  }

  80% {
    transform: translateY(0);
  }

  100% {
    transform: translateY(-18px);
  }
}
