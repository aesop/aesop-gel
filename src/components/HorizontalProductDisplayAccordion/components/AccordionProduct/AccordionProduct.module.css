@import 'styles/index';

.accordionProduct {
  position: relative;
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  transition: width 1s 0.4s;
  &.compressed {
    width: 10%;
    cursor: pointer;
  }
  @media (--viewport-sm-only) {
    padding: 0 20px;
  }
  img {
    width: 50%;
    height: auto;
  }
}

.productImage {
  width: auto;
  max-width: 100%;
  height: auto;
  max-height: 100%;
}

.accordionDefaultContent {
  position: absolute;
  z-index: 1;
  top: 0;
  display: flex;
  overflow: hidden;
  height: 100%;
  align-items: center;
  justify-content: center;
  transition: opacity 0.4s 1.4s, height 0s;
  &.hidden {
    height: 0;
    opacity: 0;
    transition: opacity 0.4s, height 0s 1.4s;
  }
  & > div {
    padding: 20px;
    text-align: center;
  }
}

.accordionExpanded {
  position: absolute;
  top: 0;
  left: 0;
  display: flex;
  width: 100%;
  height: 100%;
  justify-content: center;
  opacity: 0;
  transition: opacity 0.4s 0s;

  /* & > div {
    display: flex;
    height: 100%;
    box-sizing: border-box;
    align-items: center;
    padding: 20px;
    transform: translate3d(0, 25%, 0);
    transition: transform 0s 1.4s;
    @media (--viewport-md-only) {
      flex-direction: column;
    }
  } */
  &.open {
    opacity: 1;
    transition: opacity 0.4s 1.4s;

    /* & > div {
      transform: translate3d(0, 0, 0);
      transition: transform 2s 0.4s ease;
    } */
  }

  /* & > div > div:first-child {
    width: 30%;
    @media (--viewport-md-only) {
      position: relative;
      width: 100%;
      flex: 1;
      img {
        position: absolute;
        top: 0;
        left: 0;
        width: 100%;
        height: 100%;
        object-fit: contain;
        object-position: center bottom;
      }
    }
  } */
}

.accordionExpandedContent {
  display: flex;
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  align-items: center;
  padding: 20px;
  transform: translate3d(0, 25%, 0);
  transition: transform 0s 1.4s;
  @media (--viewport-sm-only) {
    flex-direction: column;
  }
}

.open .accordionExpandedContent {
  transform: translate3d(0, 0, 0);
  transition: transform 2s 0.4s ease;
}

.expandedAccordionImage {
  position: relative;
  width: 30%;
  @media (--viewport-sm-only) {
    width: 100%;
    flex: 1;
    img {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      object-fit: contain;
      object-position: center bottom;
    }
  }
}

.closeBtn {
  position: absolute;
  z-index: 1;
  top: 20px;
  right: 20px;
  margin: 0;
}

.openBtn {
  margin: 1em 0 2em;
}

.openBtn,
.closeBtn {
  padding: 0;
  border-top: none;
  border-right: none;
  border-bottom: 1px solid;
  border-left: none;
  background: none;
  color: var(--color-dark-copy);
  cursor: pointer;
  font-family: var(--font-family-sans-serif-regular);
  &:focus {
    outline: 0;
  }
  &.light {
    color: var(--color-light-copy);
  }
}

.accordionDefaultBackground {
  height: 100%;
  transition: opacity 0.4s 1.4s;
  &.hidden {
    opacity: 0;
    transition: opacity 0.4s 0s;
  }
}

.accordionOpenCopy,
.accordionClosedCopy {
  padding: 0;
  margin: 0 0 1em;
  font-family: var(--font-family-sans-serif-regular);
  @extend %font-theme-dark-copy;
  &.light {
    @extend %font-theme-light-copy;
  }
}

.btns {
  display: flex;
  margin: 1em 0 0;
  button,
  a {
    flex: 1;
  }
  button {
    margin: 0 4px 0 0;
  }
  @media (--viewport-sm-only) {
    display: block;
    button {
      margin: 0 0 4px;
    }
  }
}
