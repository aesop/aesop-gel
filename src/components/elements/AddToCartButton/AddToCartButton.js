import React from 'react';
import cx from 'classnames';
import Button from '~/components/elements/Button';
import Loading from '~/components/elements/Loading';
import PROP_TYPES from './AddToCartButton.prop-types';
import styles from './AddToCartButton.module.css';

const AddToCartButton = ({
  className,
  dataTestRef,
  handleOnClick,
  isDisabled = false,
  isLoading = false,
  name,
  price,
  sku,
  loading,
}) => {
  const classSet = cx(styles.base, className);

  // TODO - need contentful translations for these
  const cartActionCopy = `Add to your cart â€” ${price}`;
  const notificationCopy = `${name} added to your cart`;
  const successCopy = `Added to your cart`;

  // @TODO Mock const
  const createCartEntry = () => {};
  const data = {};
  const error = {};

  const updateSuccessful = !loading && data && data.createCartEntry;
  const labelClassName = cx(styles.label, {
    [styles.showSuccessMessage]: updateSuccessful,
  });

  return (
    <Button
      alternate={true}
      className={classSet}
      dataTestRef={dataTestRef}
      disabled={loading || isDisabled}
      onClick={handleOnClick}
      title="Add to cart button"
    >
      {isLoading ? (
        <Loading
          className={styles.loading}
          data-test-ref={`${dataTestRef}_UPDATING`}
          isLight={true}
          isLoading={true}
        />
      ) : (
        <span
          className={labelClassName}
          data-test-ref={
            updateSuccessful ? `${dataTestRef}_SUCCESS` : undefined
          }
        >
          <span>{successCopy}</span>
          <span>{cartActionCopy}</span>
        </span>
      )}
    </Button>
  );
};

AddToCartButton.propTypes = PROP_TYPES;

export default AddToCartButton;
