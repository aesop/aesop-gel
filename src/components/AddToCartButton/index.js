/* istanbul ignore file */
import AddToCartButton from './AddToCartButton';

export default AddToCartButton;
