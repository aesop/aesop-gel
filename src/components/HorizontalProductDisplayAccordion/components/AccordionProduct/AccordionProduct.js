import React from 'react';
import cx from 'classnames';
import { HYPERLINK_STYLE_TYPES } from '~/constants';
import PropTypes from 'prop-types';
import AddToCartButton from '~/components/AddToCartButton';
import Video from '~/components/Video';
import Heading from '~/components/Heading';
import Hyperlink from '~/components/Hyperlink';
import Image from '~/components/Image';
import styles from './AccordionProduct.module.css';

const AccordionProduct = ({
  addToCart,
  closedState,
  isCompressed,
  isExpanded,
  id,
  index,
  openState,
  resetAccordion,
  toggleAccordion,
}) => {
  return (
    <div
      className={cx(
        styles.accordionProduct,
        isExpanded && styles.isExpanded,
        isCompressed && styles.isCompressed,
      )}
      onClick={isCompressed ? () => resetAccordion() : undefined}
      onKeyDown={isCompressed ? () => resetAccordion() : undefined}
      role="button"
      tabIndex={isCompressed ? 0 : -1}
    >
      <div
        className={cx(
          styles.accordionDefaultBackground,
          isExpanded && styles.hidden,
        )}
        style={{ backgroundColor: closedState?.backgroundColour }}
      >
        {closedState.background === 'Video' && (
          <Video
            className={cx(styles.closedBackgroundVideo)}
            hasAllowAudio={false}
            hasAutoplay={true}
            hasControls={false}
            hasLoop={true}
            isMuted={true}
            {...closedState.backgroundVideo}
          />
        )}
        {closedState.background === 'Image' && (
          <Image
            className={cx(styles.closedBackgroundImage)}
            {...closedState.backgroundImage}
          />
        )}
      </div>

      <div
        className={cx(
          styles.accordionDefaultContent,
          isExpanded || isCompressed ? styles.hidden : '',
        )}
      >
        <div>
          {closedState?.title && (
            <Heading level="2" size="xLarge">
              {closedState?.title}
            </Heading>
          )}
          {closedState?.eyebrow && (
            <Heading level="4" size="xXSmall">
              {closedState?.eyebrow}
            </Heading>
          )}
          <div className={cx(styles.accordionClosedCopy, styles.theme)}>
            {closedState?.copy}
          </div>
          <button
            className={cx(styles.openBtn)}
            onClick={() => toggleAccordion(index, true)}
          >
            {closedState?.openButtonText}
          </button>
          <Image
            className={cx(styles.closedForegroundImage)}
            {...closedState.foregroundImage}
          />
        </div>
      </div>

      <div
        className={cx(styles.accordionExpanded, isExpanded ? styles.open : '')}
        style={{ backgroundColor: openState?.backgroundColour }}
      >
        {openState.background === 'Video' && (
          <Video
            className={cx(styles.openBackgroundVideo)}
            hasAllowAudio={false}
            hasAutoplay={true}
            hasControls={false}
            hasLoop={true}
            isMuted={true}
            {...openState.backgroundVideo}
          />
        )}
        {openState.background === 'Image' && (
          <Image
            className={cx(styles.openBackgroundImage)}
            {...openState.backgroundImage}
          />
        )}
        <div className={cx(styles.accordionExpandedContent)}>
          <div className={cx(styles.expandedAccordionImage)}>
            <Image {...openState.foregroundImage} />
          </div>
          <div className={cx(styles.expandedAccordionText)}>
            {openState?.title && (
              <Heading level="2" size="xLarge">
                {openState?.title}
              </Heading>
            )}
            {openState?.eyebrow && (
              <Heading level="4" size="xXSmall">
                {openState?.eyebrow}
              </Heading>
            )}
            <div className={cx(styles.accordionOpenCopy, styles.theme)}>
              {openState?.copy}
            </div>
            <div className={cx(styles.buttonGroup)}>
              <AddToCartButton copy={addToCart} theme={openState.theme} />
              {openState?.cta && (
                <Hyperlink
                  className={styles.cta}
                  isAlternate={false}
                  style={HYPERLINK_STYLE_TYPES.INTERNAL_BUTTON_LINK}
                  url={openState?.cta.url}
                >
                  {openState?.cta.text}
                </Hyperlink>
              )}
            </div>
          </div>
        </div>
        <button
          className={cx(styles.closeBtn)}
          onClick={() => toggleAccordion(index, false)}
        >
          {openState.closeButtonText}
        </button>
      </div>
    </div>
  );
};

AccordionProduct.propTypes = {
  addToCart: PropTypes.shape({
    cartAction: PropTypes.string,
    updateNotification: PropTypes.string,
    outOfStock: PropTypes.shape({
      label: PropTypes.string,
      title: PropTypes.string,
    }),
  }),
  closedState: PropTypes.shape({
    background: PropTypes.oneOf(['Colour', 'Image', 'Video']),
    backgroundColour: PropTypes.string,
    backgroundImage: PropTypes.object,
    backgroundVideo: PropTypes.object,
    copy: PropTypes.node,
    eyebrow: PropTypes.string,
    foregroundImage: PropTypes.object,
    id: PropTypes.string,
    openButtonText: PropTypes.string,
    theme: PropTypes.string,
    title: PropTypes.string,
  }),
  isCompressed: PropTypes.bool,
  isExpanded: PropTypes.bool,
  id: PropTypes.string,
  index: PropTypes.number,
  openState: PropTypes.shape({
    background: PropTypes.oneOf(['Colour', 'Image', 'Video']),
    backgroundColour: PropTypes.string,
    backgroundImage: PropTypes.object,
    backgroundVideo: PropTypes.object,
    closeButtonText: PropTypes.string,
    copy: PropTypes.node,
    cta: PropTypes.shape({
      text: PropTypes.string,
      url: PropTypes.string,
    }),
    eyebrow: PropTypes.string,
    foregroundImage: PropTypes.object,
    theme: PropTypes.string,
    title: PropTypes.string,
    product: PropTypes.object,
  }),
  resetAccordion: PropTypes.func,
  toggleAccordion: PropTypes.func,
};

AccordionProduct.defaultProps = {
  addToCart: {
    cartAction: undefined,
    updateNotification: undefined,
    outOfStock: {
      label: undefined,
      title: undefined,
    },
  },
  closedState: {
    background: undefined,
    backgroundColour: undefined,
    backgroundImage: undefined,
    backgroundVideo: undefined,
    copy: undefined,
    eyebrow: undefined,
    foregroundImage: undefined,
    theme: 'dark',
    title: undefined,
  },
  isCompressed: false,
  isExpanded: false,
  id: undefined,
  index: undefined,
  openState: {
    background: undefined,
    backgroundColour: undefined,
    backgroundImage: undefined,
    backgroundVideo: undefined,
    copy: undefined,
    eyebrow: undefined,
    foregroundImage: undefined,
    theme: 'dark',
    title: undefined,
  },
  resetAccordion: undefined,
  toggleAccordion: undefined,
};

export default AccordionProduct;
