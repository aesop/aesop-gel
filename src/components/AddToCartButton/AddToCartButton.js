import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { useAddToCartContext, useVariantSelectContext } from '~/contexts';
import { HYPERLINK_STYLE_TYPES } from '~/constants';
import { Button } from '~/components/Button';
import { Loading } from '~/components/Loading';
import { Hyperlink } from '~/components/Hyperlink';
import styles from './AddToCartButton.module.css';

const AddToCartButton = ({
  className,
  copy,
  dataTestRef,
  isEnabled,
  isFullWidth,
  theme,
}) => {
  const addToCartContext = useAddToCartContext();
  const { selectedVariant } = useVariantSelectContext();

  if (!selectedVariant) return null;

  const {
    alternateAction,
    isInStock,
    isSellable,
    price,
    sku,
  } = selectedVariant;

  const classSet = cx(
    styles.base,
    { [styles.fullWidth]: isFullWidth },
    className,
  );

  const handleOnClick = () => {
    const { actionTypes, dispatch, onClick } = addToCartContext;

    onClick(sku, dispatch, actionTypes);
  };

  if (!!alternateAction?.url && !!alternateAction?.label) {
    return (
      <Hyperlink
        className={classSet}
        dataTestRef={dataTestRef}
        isAlternate={true}
        style={HYPERLINK_STYLE_TYPES.INTERNAL_NO_ICON_BUTTON_LINK}
        theme={theme}
        title={alternateAction?.label}
        url={alternateAction?.url}
      >
        {alternateAction?.label}
      </Hyperlink>
    );
  }

  const {
    errorMessage,
    hasError,
    isLoading,
    isUpdateSuccessful,
  } = addToCartContext;

  if (hasError) {
    /** @TODO Handle errors thrown by handleOnClick */
    console.error('Add To Cart button updateError: ', errorMessage); // eslint-disable-line
  }

  const cartActionLabel = `${copy.cartAction} â€” ${price}`;
  const updateNotificationLabel = copy.updateNotification;
  const shouldShowUpdateSuccessMessage = !isLoading && isUpdateSuccessful;

  const isButtonEnabled =
    !isLoading &&
    !!price &&
    !!sku &&
    isInStock &&
    !!sku &&
    isEnabled &&
    !hasError &&
    isSellable;

  const labelClassName = cx(
    styles.label,
    { [styles.hideLabel]: isLoading },
    { [styles.showSuccessMessage]: shouldShowUpdateSuccessMessage },
  );

  return (
    <Button
      className={classSet}
      dataTestRef={dataTestRef}
      isAlternate={true}
      isEnabled={isButtonEnabled}
      onClick={handleOnClick}
      theme={theme}
      title={isInStock ? cartActionLabel : copy.outOfStock?.title}
    >
      {isLoading && (
        <Loading
          className={styles.loading}
          data-test-ref={`${dataTestRef}_UPDATING`}
          isLoading={true}
          theme={theme === 'dark' ? 'light' : 'dark'}
        />
      )}
      <span
        className={labelClassName}
        data-test-ref={
          shouldShowUpdateSuccessMessage ? `${dataTestRef}_SUCCESS` : undefined
        }
      >
        <span>{updateNotificationLabel}</span>
        <span>{isInStock ? cartActionLabel : copy.outOfStock?.label}</span>
      </span>
    </Button>
  );
};

AddToCartButton.propTypes = {
  className: PropTypes.string,
  copy: PropTypes.shape({
    cartAction: PropTypes.string,
    updateNotification: PropTypes.string,
    outOfStock: PropTypes.shape({
      label: PropTypes.string,
      title: PropTypes.string,
    }),
  }),
  dataTestRef: PropTypes.string.isRequired,
  isEnabled: PropTypes.bool,
  isFullWidth: PropTypes.bool,
  theme: PropTypes.oneOf(['dark', 'light']),
};

AddToCartButton.defaultProps = {
  className: undefined,
  copy: {
    cartAction: undefined,
    updateNotification: undefined,
    outOfStock: {
      label: undefined,
      title: undefined,
    },
  },
  dataTestRef: 'ADD_TO_CART',
  isEnabled: true,
  isFullWidth: true,
  theme: 'dark',
};

export { AddToCartButton };
