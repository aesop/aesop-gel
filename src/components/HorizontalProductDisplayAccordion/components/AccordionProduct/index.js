/* istanbul ignore file */
import { AccordionProduct } from './AccordionProduct';

export { AccordionProduct };
