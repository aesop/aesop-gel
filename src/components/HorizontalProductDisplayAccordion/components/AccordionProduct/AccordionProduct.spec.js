import React from 'react';
import { render } from '@testing-library/react';
import {
  AddToCartContextProvider,
  VariantSelectContextProvider,
} from '~/contexts';
import { mockAddToCartButtonOnClick } from '~/components/AddToCartButton/__mocks__/AddToCartButton.onClick';
import AccordionProductFixture from './AccordionProduct.fixture.js';
import AccordionProduct from './AccordionProduct';

describe('<AccordionProduct />', () => {
  it('should be defined', () => {
    expect(AccordionProduct).toBeDefined();
  });

  it('renders base component correctly', () => {
    const { container } = render(
      <AddToCartContextProvider onClick={mockAddToCartButtonOnClick}>
        <VariantSelectContextProvider
          variants={AccordionProductFixture.openState.product.variants}
        >
          <AccordionProduct
            addToCart={AccordionProductFixture.addToCart}
            closedState={AccordionProductFixture.closedState}
            id={AccordionProductFixture.id}
            openState={AccordionProductFixture.openState}
          />
        </VariantSelectContextProvider>
      </AddToCartContextProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
