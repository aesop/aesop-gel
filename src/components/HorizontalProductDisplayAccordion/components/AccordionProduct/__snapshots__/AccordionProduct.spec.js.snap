// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionProduct /> renders base component correctly 1`] = `
<div
  className="accordionProduct"
  role="button"
  tabIndex={-1}
>
  <div
    className="accordionDefaultBackground"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <picture
      className="base picture dark closedBackgroundImage"
    >
      <source
        media="(min-width: 1025px)"
        srcSet="./assets/images/HorizontalProductDisplayAccordion/Ptolemy-front.png"
      />
      <source
        media="(min-width: 640px)"
        srcSet="./assets/images/HorizontalProductDisplayAccordion/Ptolemy-front.png"
      />
      <source
        media="(min-width: 0px)"
        srcSet="./assets/images/HorizontalProductDisplayAccordion/Ptolemy-front.png"
      />
      <img />
    </picture>
  </div>
  <div
    className="accordionDefaultContent"
  >
    <div>
      <h2
        className="base xLarge dark"
      >
        Ptolemy
      </h2>
      <h4
        className="base xXSmall dark"
      >
        Cardamon Mimosa Tobacco
      </h4>
      <div
        className="accordionClosedCopy theme"
      >
        A beam to comfort those on the high seas
      </div>
      <button
        className="openBtn"
        onClick={[Function]}
      >
        Learn more
      </button>
      <picture
        className="base picture dark closedForegroundImage"
      >
        <source
          media="(min-width: 1025px)"
          srcSet="./assets/images/HorizontalProductDisplayAccordion/candles-product.png"
        />
        <source
          media="(min-width: 640px)"
          srcSet="./assets/images/HorizontalProductDisplayAccordion/candles-product.png"
        />
        <source
          media="(min-width: 0px)"
          srcSet="./assets/images/HorizontalProductDisplayAccordion/candles-product.png"
        />
        <img
          alt="Lorem ipsum dolor sit"
        />
      </picture>
    </div>
  </div>
  <div
    className="accordionExpanded"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <picture
      className="base picture dark openBackgroundImage"
    >
      <source
        media="(min-width: 1025px)"
        srcSet="https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM"
      />
      <source
        media="(min-width: 640px)"
        srcSet="https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM"
      />
      <source
        media="(min-width: 0px)"
        srcSet="https://www.aesop.com/medias/GelCleanser-640x640.png?context=bWFzdGVyfGltYWdlc3w1MzE5NTl8aW1hZ2UvcG5nfGltYWdlcy9oOTUvaGNkLzg4MDc3OTc4NTAxNDIucG5nfGM3NTNhYjdkMWNiOTRhNWZhNDA0NWExZWJiODMxN2VhNzEyY2ZkNGUyYzc4ZWVkZDRjYTIzMzZmZTRlOGYxNjM"
      />
      <img />
    </picture>
    <div
      className="accordionExpandedContent"
    >
      <div
        className="expandedAccordionImage"
      >
        <picture
          className="base picture dark"
        >
          <source
            media="(min-width: 1025px)"
            srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
          />
          <source
            media="(min-width: 640px)"
            srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
          />
          <source
            media="(min-width: 0px)"
            srcSet="https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ"
          />
          <img
            alt="Lorem ipsum dolor sit"
          />
        </picture>
      </div>
      <div
        className="expandedAccordionText"
      >
        <h2
          className="base xLarge dark"
        >
          Aganice
        </h2>
        <h4
          className="base xXSmall dark"
        >
          Cardamon Mimosa Tobacco
        </h4>
        <div
          className="accordionOpenCopy theme"
        >
          A beam to comfort those on the high seas
        </div>
        <div
          className="buttonGroup"
        >
          <button
            className="base alternate blockStyle dark base fullWidth"
            data-test-ref="ADD_TO_CART"
            disabled={false}
            onClick={[Function]}
            title="undefined — $26.45"
            type="button"
          >
            <span
              className="label"
            >
              <span />
              <span>
                undefined — $26.45
              </span>
            </span>
          </button>
          <a
            className="base blockStyle hasIcon left dark cta"
            href="/"
            target="_self"
          >
            Lorem ipsum dolor
             
            <i
              aria-hidden="true"
              className="icon"
            >
              <svg
                className="base rightArrow dark"
                data-ref=""
                focusable="false"
                height={12}
                role="img"
                style={
                  Object {
                    "height": "12px",
                    "width": "12px",
                  }
                }
                tabIndex={-1}
                viewBox="0 0 50 50"
                width={12}
              >
                <g>
                  <path
                    d="M30.1,5.3L50,25.1L30.1,45h-6.6l18-17.6H0v-4.8h41.5l-18-17.6h6.6V5.3z"
                  />
                </g>
              </svg>
            </i>
          </a>
        </div>
      </div>
    </div>
    <button
      className="closeBtn"
      onClick={[Function]}
    >
      Close
    </button>
  </div>
</div>
`;
